### ISSUE TITLE
ArgumentError (missing keywords: :event_type, :event_id, :data, :metadata):

### ISSUE THREAD

**Issue by remoteweb** [palkan/active_event_store#12]
## What did you do?
Implementing active event store for async events with sidekiq as ActiveJob adapter.

## What did you expect to happen?
The SubscriberJob to execute properly

## What actually happened?
ArgumentError (missing keywords: :event_type, :event_id, :data, :metadata): upon subscriber sidekiq job execution

## Additional context
<img width="1236" alt="image" src="https://github.com/palkan/active_event_store/assets/2930601/9dd56eff-9b0f-4bb9-9d31-ecab09ff2720">

## Environment

**Ruby Version: 3.2.2**

**Framework Version (Rails, whatever): Rails 7.0.7**

**Active Event Store Version: active_event_store (1.0.2)**

**Rails Event Store Version: rails_event_store (2.11.1)**

**Sidekiq Version: sidekiq (7.1.2)**

**Comment by remoteweb** [comment]
I used rails_event_store  version 2.1 and the problem disappeared. 

Something might have changed with the deserialize method of Rails Event Store gem and it causes errors in Active Event Store job execution.

**Comment by remoteweb** [comment]
Full error backtrace
```
ERROR: Error performing OnHubCompanyUpdated::SyncCompany::SubscriberJob (Job ID: 6f094d2d-a383-4c48-94dc-00f63ada7221) from Sidekiq(events_subscribers) in 185291.56ms: ArgumentError (missing keywords: :event_type, :event_id, :data, :metadata):
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/ruby_event_store-2.11.1/lib/ruby_event_store/client.rb:273:in `deserialize'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/active_event_store-1.0.2/lib/active_event_store/subscriber_job.rb:42:in `perform'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/execution.rb:65:in `block in _perform_job'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/marginalia-1.11.1/lib/marginalia/railtie.rb:32:in `block (2 levels) in insert_into_active_job'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `instance_exec'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/i18n-1.14.1/lib/i18n.rb:322:in `with_locale'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `instance_exec'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/core_ext/time/zones.rb:65:in `use_zone'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/timezones.rb:9:in `block (2 levels) in <module:Timezones>'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `instance_exec'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:138:in `run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/execution.rb:64:in `_perform_job'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/instrumentation.rb:20:in `_perform_job'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/execution.rb:52:in `perform_now'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/instrumentation.rb:14:in `block in perform_now'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/instrumentation.rb:25:in `block in instrument'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/notifications.rb:206:in `block in instrument'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/notifications.rb:206:in `instrument'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/instrumentation.rb:35:in `instrument'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/instrumentation.rb:14:in `perform_now'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/logging.rb:18:in `block in perform_now'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/tagged_logging.rb:99:in `block in tagged'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/tagged_logging.rb:37:in `tagged'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/tagged_logging.rb:99:in `tagged'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/logging.rb:25:in `tag_logger'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/logging.rb:18:in `perform_now'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/execution.rb:30:in `block in execute'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/railtie.rb:54:in `block (4 levels) in <class:Railtie>'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/execution_wrapper.rb:92:in `wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/reloader.rb:72:in `block in wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/execution_wrapper.rb:88:in `wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/reloader.rb:71:in `wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/railtie.rb:53:in `block (3 levels) in <class:Railtie>'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `instance_exec'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/callbacks.rb:138:in `run_callbacks'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/execution.rb:28:in `execute'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activejob-7.0.7/lib/active_job/queue_adapters/sidekiq_adapter.rb:42:in `perform'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:209:in `execute_job'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:177:in `block (2 levels) in process'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/middleware/chain.rb:180:in `traverse'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/middleware/chain.rb:183:in `block in traverse'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/exception_hunter-2e94efd6962f/lib/exception_hunter/middleware/sidekiq_hunter.rb:19:in `call'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/middleware/chain.rb:182:in `traverse'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/middleware/chain.rb:183:in `block in traverse'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/metrics/tracking.rb:26:in `track'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/metrics/tracking.rb:122:in `call'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/middleware/chain.rb:182:in `traverse'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/middleware/chain.rb:173:in `invoke'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:176:in `block in process'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:140:in `block (6 levels) in dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/job_retry.rb:114:in `local'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:139:in `block (5 levels) in dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/rails.rb:16:in `block in call'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/execution_wrapper.rb:92:in `wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/reloader.rb:72:in `block in wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/execution_wrapper.rb:92:in `wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/activesupport-7.0.7/lib/active_support/reloader.rb:71:in `wrap'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/rails.rb:15:in `call'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:135:in `block (4 levels) in dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:270:in `stats'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:130:in `block (3 levels) in dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/job_logger.rb:13:in `call'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:129:in `block (2 levels) in dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/job_retry.rb:81:in `global'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:128:in `block in dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/job_logger.rb:39:in `prepare'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:127:in `dispatch'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:175:in `process'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:82:in `process_one'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/processor.rb:72:in `run'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/component.rb:10:in `watchdog'
/Users/arisxama/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sidekiq-7.1.2/lib/sidekiq/component.rb:19:in `block in safe_thread'
2023-08-18T12:07:51.359Z pid=34729 tid=4b6t class=OnHubCompanyUpdated::SyncCompany::SubscriberJob jid=76a3177eb4a9071d664c81b1 elapsed=185.342 INFO: fail
```

**Comment by remoteweb** [comment]
The problem is this part of code here

```
# .rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/active_event_store-1.0.2/lib/active_event_store/subscriber_job.rb#41
event = event_store.deserialize(**payload, serializer: ActiveEventStore.config.serializer)
```

Payload from Rails Event Store 2.11.1 is
```
 {"event_id"=>"7eac442a-6966-4241-a33f-419dd82f3a50",
 "data"=>"\"{\\\"company_guid\\\":\\\"UUID\\\"}\"",
 "metadata"=>"\"{\\\"timestamp\\\":\\\"2023-08-18 12:58:27 UTC\\\",\\\"valid_at\\\":\\\"2023-08-18 12:58:27 UTC\\\",\\\"correlation_id\\\":\\\"1a56cc7f-fa95-47d7-96bb-36db02bd8181\\\"}\"",
 "event_type"=>"hub_company_updated",
 "timestamp"=>"2023-08-18T12:58:27.114715Z",
 "valid_at"=>"2023-08-18T12:58:27.114715Z"}
```

while Payload from Rails Event Store 2.1 was
```
{:event_id=>"7eac442a-6966-4241-a33f-419dd82f3a50",
 :data=>"\"{\\\"company_guid\\\":\\\"UUID\\\"}\"",
 :metadata=>"\"{\\\"timestamp\\\":\\\"2023-08-18 12:58:27 UTC\\\",\\\"valid_at\\\":\\\"2023-08-18 12:58:27 UTC\\\",\\\"correlation_id\\\":\\\"1a56cc7f-fa95-47d7-96bb-36db02bd8181\\\"}\"",
 :event_type=>"hub_company_updated",
 :timestamp=>"2023-08-18T12:58:27.114715Z",
 :valid_at=>"2023-08-18T12:58:27.114715Z"}
```

due to this part of code that changed in Rails Event Store gem
```
# .rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rails_event_store-2.11.1/lib/rails_event_store/active_job_scheduler.rb#12
klass.perform_later(record.serialize(serializer).to_h.transform_keys(&:to_s))
```

```
# .rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rails_event_store-2.1.0/lib/rails_event_store/active_job_scheduler.rb#12
klass.perform_later(record.serialize(serializer).to_h)
```

**Comment by palkan** [comment]
Thanks for the detailed report. Will take a look later this week.

**Comment by palkan** [comment]
Fixed and released in 1.1.0.
